feat(dapps) check Wallet Connect pair uri for already used and expired

stefandunca opened the pull request:
### Updates: #14676

https://github.com/status-im/status-desktop/assets/47554641/0e6ac19e-b292-454c-b3c1-d6696fd259d2

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d43ae54d | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-15468/1/) | 2024-07-05 09:40:54 | ~6 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-15468/1//consoleText) |

caybro commented on ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml:
```diff
@@ -151,8 +155,10 @@ DappsComboBox {
                 function onMaxFeesUpdated(maxFees, maxFeesWei, haveEnoughFunds, symbol) {
                     maxFeesText = `${maxFees.toFixed(2)} ${symbol}`
```
This should be formatted using `LocaleUtils`; kinda hard to do with the `ethStr` below... :thinking:

stefandunca replied:
This is just a dummy display code to show that the data is coming from the SDK. It has to be replaced with the proper status-go based implementation here in this work package: https://github.com/status-im/status-desktop/issues/15192

caybro replied:
Ah cool

caybro reviewed (approved):
LGTM otherwise

alexjba commented:
Now I see we're missing the flow where if the url is valid we should automatically move forward and open the connect dialog. Do you mind adding this here? Or we can create another task so that we don't forget about it.

alexjba commented on ui/imports/shared/popups/walletconnect/PairWCModal/WCUriInput.qml:
WDYT of keeping all the validation rules in a single place?
Now it's split between `WCUriInput` and `WalletConnectService`

alexjba commented on ui/imports/shared/popups/walletconnect/PairWCModal/WCUriInput.qml:
I think now we can remove this
```
        function wcUriAlreadyUsed(uri) {
            // TODO: Check if URI is already used
            return false
        }
```

alexjba reviewed (approved):
Nice! There is some code we need to remove here.
Other than that, LGTM!

alexjba commented:
qml tests tail with `SwapInputPanel ` error. I've thought we fixed this. Is this an old branch?
```
[2024-07-07T13:16:55.175Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Compared values are not the same
[2024-07-07T13:16:55.175Z] 3:    Actual   (): 0
[2024-07-07T13:16:55.175Z] 3:    Expected (): 5.42
[2024-07-07T13:16:55.175Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15468/storybook/qmlTests/tests/tst_SwapInputPanel.qml(424)]
```

caybro commented:
> qml tests tail with `SwapInputPanel ` error. I've thought we fixed this. Is this an old branch?
> 
> ```
> [2024-07-07T13:16:55.175Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Compared values are not the same
> [2024-07-07T13:16:55.175Z] 3:    Actual   (): 0
> [2024-07-07T13:16:55.175Z] 3:    Expected (): 5.42
> [2024-07-07T13:16:55.175Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15468/storybook/qmlTests/tests/tst_SwapInputPanel.qml(424)]
> ```

Flaky test, I've just seen it too, restart helped

stefandunca commented:
Done, Thanks for the great input

> Now I see we're missing the flow where if the url is valid we should automatically move forward and open the connect dialog. Do you mind adding this here? Or we can create another task so that we don't forget about it.

I implemented and tested it within this PR

> WDYT of keeping all the validation rules in a single place?

I moved them all in Service, maybe we could even get further in a follow up change and group all the pairing logic under the sme file/object.
